### ISSUE TITLE
Fix login form

### ISSUE THREAD

**Issue by raygdev** [raygdev/simply-lit#55]
Currently the login form is slightly broken. Namely the login action errors.

**Comment by CampbellSDC** [comment]
Looking over the login form, I see a couple issues:

<img width="875" alt="Screen Shot 2023-11-16 at 10 53 13 AM" src="https://github.com/raygdev/simply-lit/assets/89622690/09294726-e2c1-4033-869b-5297b260c2b8">

The action function in the login form needs a ```redirect``` from the login form screen to the user dashboard. 

There also needs to be a login route created for the user email and password to be sent to Mongo. I've only done this with Firebase, so I'll look up how to do that. I created a 'api.js' file on the client side to store that 'userLogin' function. Am I on the right path here? Sorry, it's been crazy trying to keep up with everything here, at RefCode and just at home with holiday stuff. 

How are things over there? You added so much stuff last week lol. I'll try to do the same this week.

**Comment by raygdev** [comment]
@CampbellSDC yeah, you are spot on. We do need a redirect. The dashboard route is going to need to request the user. 

The login route is already created on the server. It is `"/login"`.  Once the credentials are verified with the username and password, the redirect should happen for dashboard... which we still need to create 😆! Any information we need from the user will be from that root dashboard loader. 

1) user inputs information into login form.

2) it navigates to the login action.

3) if we get a successful 200 response, we can navigate to dashboard. 

4) if response is 404 we return the message from the server which should be an object that states the email/password combination doesn't exist.

5) dashboard is going to request the user information in it's loader. This will be the root loader we use for the user information and roles.

I have to set up a verification endpoint to just ping the server to see if the user is still authenticated. I think I may have already done that, but that will be out endpoint to ensure the user is authenticated in any loader for the protected route. It will simply return a 200 or 401. If 200 we are good, if 401 redirect back to login.

I did do a bit last week 😄! But I tried to commit in stages so that it would be easier for you to navigate. You can always go back to that closed pull request and go through each commit to kind of walk through the code. If you have any questions, I can certainly help with that. I tried to commit in order so that there would be a good logical flow of building that stuff up. So hopefully it makes chronological sense. 

Yeah holidays are always busy. I still need to get to the store for food before it all disappears 😆! I don't want to do last minute and have everything wiped out! What are you all doing for Thanksgiving? Staying home? Going out? Going to a family gathering?
